tests_and_analysis/test/euphonic_test/test_force_constants_calculate_qpt_phonons_castep.py

```diff
@@ -0,0 +1,492 @@
+        [0.00, 0.00, 0.50],
+        [0.00, 0.00, 0.00],
+        [0.00, 0.00, 0.00],
+        [-0.25, 0.50, 0.50],
+        [0.00, 0.00, 0.00],
+        [0.00, 0.00, 0.00],
+        [0.00, 0.00, 0.00],
+        [0.00, 0.00, 0.00],
+        [0.00, 0.00, 0.00],
+        [0.00, 0.00, 0.00],
+        [-0.151515, 0.575758, 0.5],
+        [0.00, 0.00, 0.00]
+    ])
+    split_qpts_insert_gamma = np.array([
+        [0.00, 0.00, 0.00],
+        [0.00, 0.00, 0.50],
+        [0.00, 0.00, 0.00],
+        [-0.25, 0.50, 0.50],
+        [0.00, 0.00, 0.00],
+        [0.00, 0.00, 0.00],
+        [0.00, 0.00, 0.00],
+        [-0.151515, 0.575758, 0.5],
+        [0.00, 0.00, 0.00]
+    ])
+
+    quartz_test_data = [
+        (
+            {"qpts": qpts, "dipole": True, "splitting": False},
+            2e-6
+        ),
+        (
+            {
+                "qpts": qpts, "dipole": True, "splitting": False,
+                "use_c": True, "fall_back_on_python": False
+            },
+            2e-6
+        ),
+        (
+            {
+                "qpts": qpts, "dipole": True, "splitting": False, "use_c": True,
+                "n_threads": 2, "fall_back_on_python": False
+            },
+            8e-8
+        ),
+        (
+            {
+                "qpts": qpts, "asr": 'reciprocal',
+                "dipole": True, "splitting": False
+            },
+            5e-4
+        ),
+        (
+            {
+                "qpts": qpts, "asr": 'reciprocal', "dipole": True,
+                "splitting": False, "use_c": True, "fall_back_on_python": False
+            },
+            5e-4
+        ),
+        (
+            {
+                "qpts": qpts, "asr": 'reciprocal', "dipole": True,
+                "splitting": False, "use_c": True, "n_threads": 2,
+                "fall_back_on_python": False
+            },
+            5e-4
+        ),
+        (
+            {
+                "qpts": split_qpts, "asr": 'reciprocal',
+                "dipole": True, "splitting": True
+            },
+            5e-4
+        ),
+        (
+            {
+                "qpts": split_qpts, "asr": 'reciprocal', "dipole": True,
+                "splitting": True, "use_c": True, "fall_back_on_python": False
+            },
+            5e-4
+        ),
+        (
+            {
+                "qpts": split_qpts, "asr": 'reciprocal', "dipole": True,
+                "splitting": True, "use_c": True, "n_threads": 2,
+                "fall_back_on_python": False
+            },
+            5e-4
+        ),
+        (
+            {
+                "qpts": split_qpts_insert_gamma, "asr": 'reciprocal',
+                "dipole": True, "splitting": True, "insert_gamma": True
+            },
+            5e-4
+        )
+    ]
+
+    quartz_castep_bin_file = os.path.join(path, "quartz", "quartz.castep_bin")
+
+    @pytest.mark.parametrize(("kwargs", "atol"), quartz_test_data)
+    def test_quartz_calculate_qpoint_phonon_modes(self, kwargs, atol):
```

Comment:
This is testing the same function as `test_calculate_qpoint_phonon_modes` above, so we can probably combine these into the same test? Comments from `test_calculate_qpoint_phonon_modes` also apply here.

Reply:
Combined